CRM/Case/Form/Activity/LinkCases.php

```diff
@@ -145,7 +145,7 @@ public static function endPostProcess($form, $params, $activity) {
     //create a link between two cases.
     if ($activityId && $linkCaseID) {
       $caseParams = [
-        'case_id' => $linkCaseID,
+        'case_id' => array_merge($form->_caseId, [$linkCaseID]),
```

Comment:
This doesn't work because processCaseActivity is expecting an int. It ends up linking every case to case id 1. And because the test down below has been changed to just test the activity api instead of simulating the form it doesn't catch this (admittedly it wasn't a great simulation, but called the form function).

Reply:
Better unit test of the form: https://github.com/civicrm/civicrm-core/pull/24328

Reply:
Thanks @demeritcowboy but I don't think `processCaseActivity` actually gets called from here. This PR changes the Activity BAO to call a new function that can handle an array.

Reply:
But just run it and you'll see it does, and links it to case id 1.

Reply:
Ok thanks I'll have a look